core/cmd/hoverfly/main.go

```diff
@@ -283,6 +282,8 @@ func main() {
 		requestCache = nil
 	}
 
+	cfg.DisableProxyAuthoriation = *disableProxyAuth
```

Comment:
I think what @mogronalol suggested is quite clear. I don't see any point of having a `disable-standard-proxy-auth` flag. If hoverfly is started with header-auth enabled, then any attempt to use `Proxy-Authorization` header should fail.